cast/src/lib.rs

```diff
@@ -110,6 +110,60 @@ where
         })
     }
 
+    /// Generates an access list for the specified transaction
+    ///
+    /// ```no_run
+    /// 
+    /// use cast::Cast;
+    /// use ethers_core::types::{Address, Chain};
+    /// use ethers_providers::{Provider, Http};
+    /// use std::{str::FromStr, convert::TryFrom};
+    ///
+    /// # async fn foo() -> eyre::Result<()> {
+    /// let provider = Provider::<Http>::try_from("http://localhost:8545")?;
+    /// let cast = Cast::new(provider);
+    /// let to = Address::from_str("0xB3C95ff08316fb2F2e3E52Ee82F8e7b605Aa1304")?;
+    /// let sig = "function greeting(uint256 i) public returns (string)";
```

Comment:
nit - this can be simplified
```suggestion
    /// let sig = "greeting(uint256)(string)";
```